netmaster/master/netmaster.go

```diff
@@ -110,33 +110,41 @@ func CreateTenant(stateDriver core.StateDriver, tenant *intent.ConfigTenant) err
 	gCfg.Auto.VLANs = tenant.VLANs
 	gCfg.Auto.VXLANs = tenant.VXLANs
 	gCfg.Auto.AllocSubnetLen = tenant.AllocSubnetLen
-	err = gCfg.Write()
-	if err != nil {
-		log.Errorf("error updating tenant '%s'.Error: %s", tenant.Name, err)
-		return err
-	}
 
 	tempRm, err := resources.GetStateResourceManager()
 	if err != nil {
 		return err
 	}
 
+	// setup resources
 	err = gCfg.Process(core.ResourceManager(tempRm))
 	if err != nil {
 		log.Errorf("Error updating the config %+v. Error: %s", gCfg, err)
 		return err
 	}
 
+	// start skydns container
 	err = startServiceContainer(tenant.Name)
 	if err != nil {
 		log.Errorf("Error starting service container. Err: %v", err)
 		return err
 	}
 
+	err = gCfg.Write()
+	if err != nil {
+		log.Errorf("error updating tenant '%s'.Error: %s", tenant.Name, err)
+		return err
+	}
+
 	return nil
 }
 
 func startServiceContainer(tenantName string) error {
+	// do nothing in test mode
+	if testMode {
```

Comment:
hmm, not sure if this is a good idea i.e. mixing test logic with production logic.

We should explore using mocks or stubs for unit testing. For sanity/system tests we should never need to block any production code.